gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/resolver/GfxdPartitionByExpressionResolver.java

```diff
@@ -670,6 +686,9 @@ public boolean okForColocation(GfxdPartitionResolver rslvr) {
       // Minimal check do determine if both have them have expr node at least
       if (this.exprCompiler != null && passed.exprCompiler == null
           || this.exprCompiler == null && passed.exprCompiler != null) {
+        if(snappyStore){ //escaping the check for snappystore.
+          return true;
+        }
         return false;
```

Comment:
For first cut we can use the expression based partitioner. However, the above code changes bypass the expressions completely and always do the new column based partitioning. That's what my primary objection was. It should continue to use expressions with a special tag/expression for the new partitioner if required. General expression based partitioning is still useful for many cases like partitioning on month/year extracting from timestamp column but these changes will not allow that to be done (we don't allow this for column tables yet, but for row tables this is still valid).

For simplicity we can interpret PARTITION BY COLUMN when isSnappyStore=true to be the special expression I mentioned above (i.e. requiresSerializedHash=false for now) and use the deserialization approach as in your current patch, then gradually add overrides to both DVD and byte[] form for different datatypes starting from SQLChar/SQLVarchar/SQLClob (and its byte form).